src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -215,7 +215,9 @@ WindowTransform::WindowTransform(const Block & input_header_,
         }
         workspace.argument_columns.assign(f.argument_names.size(), nullptr);
 
-        workspace.window_function_impl = aggregate_function->asWindowFunction();
+        /// Currently we have slightly wrong mixup of the interfaces of Window and Aggregate functions.
+        workspace.window_function_impl = dynamic_cast<IWindowFunction *>(const_cast<IAggregateFunction *>(aggregate_function.get()));
```

Comment:
Consider to check if isOnlyWindowFunction true then use static_cast.

Reply:
Does not work for "sibling" interfaces.